### PR TITLE
Added solution

### PR DESCRIPTION
**Pull request by AndrewPolishchuc**

### CODE REVIEW

File: src/main/java/com/internet/shop/controllers/CompleteCurrentOrderController.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private OrderService orderService
+            = (OrderService) injector.getInstance(OrderService.class);
+    private ShoppingCartService shoppingCartService
+            = (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        Long cartId = Long.valueOf(req.getParameter("cartId"));
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/CompleteCurrentOrderController.java]: Let's use the USER_ID=1L until we implement authorization

Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/CompleteCurrentOrderController.java]:
```suggestion
       ShoppingCart shoppingCart = shoppingCartService.getByUserId(USER_ID);
```

File: src/main/webapp/web.xml [hunk 2]
```diff
@@ -75,4 +75,68 @@
...
+        <servlet-name>completeCurrentOrder</servlet-name>
+        <url-pattern>/order/complete</url-pattern>
+    </servlet-mapping>
+    <servlet>
+        <servlet-name>detailOfOrder</servlet-name>
+        <servlet-class>com.internet.shop.controllers.GetDetailOfOrder</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>detailOfOrder</servlet-name>
+        <url-pattern>/user/order/detail</url-pattern>
```
Comment (dariiabutorina) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/user/order/details</url-pattern>
```

**Review by dariiabutorina — CHANGES_REQUESTED** [review]

File: src/main/webapp/web.xml (hunk 2 above)
Comment (AndrewPolishchuc) [reply src/main/webapp/web.xml]: Done

File: src/main/java/com/internet/shop/controllers/CompleteCurrentOrderController.java (hunk 1 above)
Comment (AndrewPolishchuc) [reply src/main/java/com/internet/shop/controllers/CompleteCurrentOrderController.java]: Okay. Done

Comment (AndrewPolishchuc) [reply src/main/java/com/internet/shop/controllers/CompleteCurrentOrderController.java]: Done

File: src/main/java/com/internet/shop/controllers/GetDetailOfOrder.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import com.internet.shop.model.Product;
+import com.internet.shop.service.OrderService;
+import java.io.IOException;
+import java.math.BigDecimal;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetDetailOfOrder extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controllers/GetDetailOfOrder.java]:
```suggestion
public class GetOrderDetailsController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/GetDetailOfOrder.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    private BigDecimal findCurrentSumOrder(Long userId) {
+        double result = 0;
+        for (Order order : orderService.getAll()) {
+            for (Product orderProduct : order.getProducts()) {
+                result += orderProduct.getPrice().doubleValue();
+            }
+        }
+        return new BigDecimal(result);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controllers/GetDetailOfOrder.java]: let's create method in OrderService for this. And you can use Stream api

File: src/main/java/com/internet/shop/controllers/GetOrderController.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import com.internet.shop.model.Order;
+import com.internet.shop.service.OrderService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetOrderController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controllers/GetOrderController.java]:
```suggestion
public class GetAllOrdersController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/ShoppingCartController.java [hunk 6]
```diff
@@ -23,6 +23,7 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
         ShoppingCart currentShoppingCart = shoppingCartService.getByUserId(USER_ID);
         List<Product> products = currentShoppingCart.getProducts();
         req.setAttribute("allProducts", products);
+        req.setAttribute("cartId", currentShoppingCart.getId());
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controllers/ShoppingCartController.java]: why do you need this ? You can use shoppingCartService.getByUserId(USER_ID) in each controller. And you will always have the id of current user(for now it's USER_ID = 1).

File: src/main/java/com/internet/shop/controllers/GetDetailOfOrder.java (hunk 3 above)
Comment (AndrewPolishchuc) [reply src/main/java/com/internet/shop/controllers/GetDetailOfOrder.java]: Done

File: src/main/java/com/internet/shop/controllers/GetOrderController.java (hunk 5 above)
Comment (AndrewPolishchuc) [reply src/main/java/com/internet/shop/controllers/GetOrderController.java]: Done

File: src/main/java/com/internet/shop/controllers/ShoppingCartController.java (hunk 6 above)
Comment (AndrewPolishchuc) [reply src/main/java/com/internet/shop/controllers/ShoppingCartController.java]: Sure. Done

File: src/main/java/com/internet/shop/controllers/GetDetailOfOrder.java (hunk 4 above)
Comment (AndrewPolishchuc) [reply src/main/java/com/internet/shop/controllers/GetDetailOfOrder.java]: Done

File: src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java [hunk 7]
```diff
@@ -44,4 +46,11 @@ public Order get(Long id) {
     public boolean delete(Long id) {
         return orderDao.deleteById(id);
     }
+
+    @Override
+    public BigDecimal findSum(Long orderId) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java:51]: let's pass order to this method so we will not call database a second time

**Review by dmytro-hryhoruk — APPROVED** [review]

File: src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java (hunk 7 above)
Comment (AndrewPolishchuc) [reply src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java:51]: Sure

**Pull request merged**